mithril-aggregator/src/services/snapshotter/compressed_archive_snapshotter.rs

```diff
@@ -94,512 +123,255 @@ impl CompressedArchiveSnapshotter {
-        snapshot_file_tar.seek(SeekFrom::Start(0))?;
-
-        let mut snapshot_archive: Archive<Box<dyn Read>> = match self.compression_algorithm {
-            SnapshotterCompressionAlgorithm::Gzip => {
-                let snapshot_file_tar = GzDecoder::new(snapshot_file_tar);
-                Archive::new(Box::new(snapshot_file_tar))
-            }
-            SnapshotterCompressionAlgorithm::Zstandard(_) => {
-                let snapshot_file_tar = Decoder::new(snapshot_file_tar)?;
-                Archive::new(Box::new(snapshot_file_tar))
-            }
-        };
-
-        let unpack_temp_dir = self
-            .temp_dir
-            .join("mithril_snapshotter_verify_archive")
-            // Add the archive name to the directory to allow two verifications at the same time
-            // (useful for tests).
-            .join(
-                archive_path
-                    .file_name()
-                    .ok_or(SnapshotError::VerifyArchiveError(format!(
-                        "Could not append archive name to temp directory: archive `{}`",
-                        archive_path.display(),
-                    )))?,
-            );
-
-        fs::create_dir_all(&unpack_temp_dir).map_err(|e| {
-            SnapshotError::VerifyArchiveError(format!(
-                "Could not create directory `{}`: {e}",
-                unpack_temp_dir.display(),
-            ))
-        })?;
-
-        let unpack_temp_file = &unpack_temp_dir.join("unpack.tmp");
-
-        let verify_result = {
-            let mut result = Ok(());
-            for e in snapshot_archive.entries()? {
-                match e {
-                    Err(e) => {
-                        result = Err(anyhow!(SnapshotError::InvalidArchiveError(format!(
-                            "invalid entry with error: '{:?}'",
-                            e
-                        ))));
-                        break;
-                    }
-                    Ok(entry) => Self::unpack_and_delete_file_from_entry(entry, unpack_temp_file)?,
-                };
-            }
-            result
-        };
-
-        // Always remove the temp directory
-        fs::remove_dir_all(&unpack_temp_dir).map_err(|e| {
-            SnapshotError::VerifyArchiveError(format!(
-                "Could not remove directory `{}`: {e}",
-                unpack_temp_dir.display(),
-            ))
-        })?;
-
-        verify_result
-    }
+    /// Returns the list of files and directories to include in ancillary snapshot.
+    ///
+    /// The immutable file included in the ancillary archive corresponds to the last one (and not finalized yet)
+    /// when the immutable file number given to the function corresponds to the penultimate.
+    fn get_files_and_directories_for_ancillary_snapshot(
+        &self,
+        immutable_file_number: u64,
+    ) -> StdResult<Vec<PathBuf>> {
+        let mut files_to_snapshot = Vec::with_capacity(4);
```

Comment:
Just a detail if you want to be really rigorous about repsonsibilities.
This number of files suppose that we know how many files are returned from immutable_trio_names.
We assume there is 3 files because we name it 'trio' but it's only because we don't have a better name to represent that trio. 
I think we can keep the current code.

Reply:
You're right, I changed the code to avoid the confusion.